### DISCUSSION TITLE
Test case gif rendering

### DISCUSSION THREAD

**Discussion by Eldolfin** [helix-editor/helix#11548] [Ideas]
Not much of a new feature, more of an automation for documentation.

Render integration tests into GIFs, with the pressed keys shown in the bottom left corner. These GIFs could be listed on a wiki page as a feature showcase and as another way to discover movements.

Additionally, a bot could post a PR comment with new test cases rendered as GIFs to showcase the newly added feature. Ideally, the bot would edit a comment at the top each time a new commit is added for easier visibility.

Reference: 
- https://github.com/charmbracelet/vhs (which is not a perfect fit as there is no Rust API as far as I know).
- https://crates.io/crates/autocast (generate files for [asciinema](https://asciinema.org/) which is not a GIF)

**Comment by TornaxO7** [comment]
I'd actually prefer a video for demos because you can jump around in videos while you may have to wait a bit until the relevant pops up in GIFs.

**Reply by Eldolfin** [reply]
good point, and being able to pause is good too.
Asciinema would probably work fine as well to start, and the files could probably then be converted to mp4 ?
